sashimi/gui/main_gui.py

```diff
@@ -45,6 +45,7 @@ def __init__(self, st: State, style: str):
         self.wid_camera = CameraSettingsWidget(st, self.wid_display, self.timer)
         self.wid_status_bar = StatusBarWidget(st, self.timer)
         self.toolbar = TopWidget(st, self.timer)
+        self.prev_exp_state = GlobalState.PAUSED
```

Comment:
This variable is used to detect the change in the state -> like an On/Off switch
we want to detect the change so that we enable/disable the gui only at the change

Reply:
Using `current_exp_state` instead of `is_exp_running` is more consistent with your nomenclature. If you want a boolean (for other/future applications) I would add a method

```python
is_exp_running(self):
    return self.current_exp_state == GlobalState.EXPERIMENT_RUNNING
```

Of course this is more a matter of style than anything else. But I see the state structure cleaner by the use of state classes than booleans, then using booleans upstream (e.g. in GUI) makes further sense.